src/main/java/seedu/address/model/event/UniqueEventList.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.contact.exceptions.ContactNotFoundException;
+import seedu.address.model.contact.exceptions.DuplicateContactException;
+
+public class UniqueEventList implements Iterable<Event> {
+
+    private final ObservableList<Event> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Event> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent module as the given argument.
+     */
+    public boolean contains(Event toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Adds a module to the list.
+     * The module must not already exist in the list.
+     */
+    public void add(Event toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateContactException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the module {@code target} in the list with {@code editedModule}.
+     * {@code target} must exist in the list.
+     * The module identity of {@code editedModule} must not be the same as another existing module in the list.
+     */
+    public void setEvent(Event target, Event editedPodule) {
+        requireAllNonNull(target, editedPodule);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ContactNotFoundException();
+        }
+
+        if (!target.equals(editedPodule) && contains(editedPodule)) {
+            throw new DuplicateContactException();
+        }
+
+        internalList.set(index, editedPodule);
+    }
+
+    /**
+     * Removes the equivalent module from the list.
+     * The module must exist in the list.
+     */
```

Comment:
module is repeated multiple times throughout the javadocs.

Reply:
Will implement later on!